### PR TITLE
Security related functionality part two

### PR DESCRIPTION
**Pull request by BolsunovaNataliia**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 1]
```diff
@@ -28,31 +31,37 @@
...
     private final BookService bookService;
 
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
     @GetMapping
     @Operation(summary = "Get all books", description = "Get a list of all available books")
-    public List<BookDto> findAll(Pageable pageable) {
-        return bookService.findAll(pageable);
+    public List<BookDto> findAll(Authentication authentication, Pageable pageable) {
+        User user = (User) authentication.getPrincipal();
+        return bookService.findAll(user.getEmail(), pageable);
```
Comment (DPotapov93) [inline src/main/java/mate/academy/bookshop/controller/BookController.java:39]: "I'm not sure, but it seems that it's not required in our task. We just need access to all books in the store to choose them from the list."

**Review by DPotapov93 — APPROVED** [review]
Good Job!

File: src/main/java/mate/academy/bookshop/Application.java
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/Application.java:1]: Why did you rename this class?

File: src/main/java/mate/academy/bookshop/controller/AuthController.java [hunk 2]
```diff
@@ -26,4 +27,9 @@ public UserResponseDto registerUser(@RequestBody @Valid UserRegistrationRequestD
             throws RegistrationException {
         return userService.register(requestDto);
     }
+
+    @PostMapping("/login")
+    public boolean login(@RequestBody UserLoginRequestDto requestDto) {
+        return true;
+    }
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/controller/AuthController.java]: it's not for this hw.

File: src/main/java/mate/academy/bookshop/dto/UserLoginRequestDto.java
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/dto/UserLoginRequestDto.java:1]: login is not for this hw.

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 3]
```diff
@@ -30,4 +38,46 @@ public class User {
     private String shippingAddress;
     @Column(nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany
+    @JoinTable(
+            name = "users_roles",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "role_id")
+    )
+    private Set<Role> roles = new HashSet<>();
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java:51]: please exclude from `equals&hashcode`

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 4]
```diff
@@ -30,4 +38,46 @@ public class User {
...
+    }
+
+    @Override
+    public String getPassword() {
+        return password;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 5]
```diff
@@ -30,4 +38,46 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 6]
```diff
@@ -30,4 +38,46 @@ public class User {
...
+    }
+
+    @Override
+    public boolean isCredentialsNonExpired() {
+        return true;
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/User.java]:
```suggestion
        return !isDeleted;
```

File: src/main/java/mate/academy/bookshop/repository/user/UserRepository.java [hunk 7]
```diff
@@ -4,9 +4,12 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface UserRepository extends JpaRepository<User, Long>, JpaSpecificationExecutor<User> {
-    Optional<User> findByEmail(String email);
+    @Query("SELECT u FROM User u LEFT JOIN FETCH u.roles WHERE u.email = :email")
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/repository/user/UserRepository.java:13]:
```suggestion
```
you can use named query

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -1,6 +1,6 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/book_shop
 spring.datasource.username=root
-spring.datasource.password=x
+spring.datasource.password=fiasko13@3
```
Comment (d1sam) [inline src/main/resources/application.properties:3]:
```suggestion
spring.datasource.password=*****************
```

File: src/main/resources/db/changelog/changes/04-insert-books.yaml [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+                - column: { name: cover_image, value: "some link 2" }
+        - insert:
+            tableName: books
+            columns:
+              - column: { name: title, value: "some title 3" }
+              - column: { name: author, value: "some author 3" }
+              - column: { name: isbn, value: "123-4-123-4-7" }
+              - column: { name: price, value: "1550" }
+              - column: { name: description, value: "some description 3" }
+              - column: { name: cover_image, value: "some link 3" }
```
Comment (d1sam) [inline src/main/resources/db/changelog/changes/04-insert-books.yaml:32]:
```suggestion
              - column: { name: cover_image, value: "some link 3" }
              
```

File: src/main/java/mate/academy/bookshop/security/CustomUserDetailsService.java
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/security/CustomUserDetailsService.java:1]: it's not for this hw.

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but please make changes according to my comments and remove changes from another hw.

File: src/main/java/mate/academy/bookshop/Application.java
Comment (BolsunovaNataliia) [reply src/main/java/mate/academy/bookshop/Application.java:1]: To unify.

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**